### DISCUSSION TITLE
TickTick Task Notes in Obsidian.

### DISCUSSION THREAD

**Discussion by thesamim** [thesamim/TickTickSync#201] [Ideas]
Howdy folks, 

Looking for input from users on the preferred way to represent TickTick Task notes in Obsidian. Please see the options below.

My preferred implementation is the call out option ("Option 1") because I think it looks cleaner and makes the coding a lot easier. Note that callouts look different (cooler in my opinion) in Obsidian that they do in GitHub. Among other things, callouts in Obsidian can be collapsible.

"Option 2" is the Obsidian "standard" markdown implementation. (ie, any lines after the task that start with two spaces are part of the task.) It is more standard markdown but makes coding  a lot more... interesting.

Some questions:

1. Do you anticipate adding notes to subtasks?
2. Do you anticipate having checklist items inside your task notes?
3. Do you anticipate adding references to other tasks inside task notes?

In the fullness of time, both choices could be presented and a preference set. But I want to start with implementing one or the other to get it right then add on the preferences option.

# Here are the options.
Let me know your thoughts in comments.

## Option 1
- [ ] Task 1
> [!NOTE] 
> Task 1 Notes
> ## A Heading
> Some content

which would look like this without formatting:
```
- [ ] Task 1
> [!NOTE]   
> Task 1 Notes
> ## A Heading
> Some content

```

## Option 2

- [ ] Task 2
  Task 2 Notes
  ## A Heading
  Some content

which would look like this without formatting:
```
- [ ] Task 2
  Task 2 Notes
  ## A Heading
  Some content
```

**Comment by thesamim** [comment]
@ms3056, @Dennyyusuf, @jgindin, @quanru, @Lorite, @raykr, @NaturallyAsh

Would appreciate getting your feedback on the topic. Thanks.

**Comment by zuwf** [comment]
First of all thank you for the wonderful plugin 

It seamlessly integrates with my TT

Still I have some disappointment regarding aesthetic 

-  It will be really helpful if the title in ticktick is unaltered for linking with the source file instead I require the link to be in description 
- Secondly, in obsidian. It could be cool if the ticktick id is hidden with css snippets in both live preview and reading mode

Additional requirements..


A option to integrate tasks from all over vault (except project file) to be added in a single list on ticktick

I have no idea.. if these are possible or not, but I wish this..

Once again thank you.. I understand your efforts over time..

**Reply by thesamim** [reply]
Title handling please see:  #131 is on the roadmap
ID Hiding: That should already be happening. If it's not happening for you please open a new issue and let me know what plugins you have installed in addition to the TickTickSync.
Tasks to to a single list: #148, #152 and to a certain extent #96 all contribute to your ask. keep an eye on those.

**Reply by zuwf** [reply]
Thank you pal.
I'll frequently follow your updates for betterment

**Comment by Kageetai** [comment]
Hey, first also thanks for the plugin. I use both, Obsidian and TickTick a lot so combining them was always on my agenda.

I am generally for the second option, as it's cleaner and more standard conform.

But there this a third option I would prefer even more: **A file per task**
Meaning, each task in TickTick gets its own file in Obsidian. Also, if I create a new file in Obsidian in a specified folder, it should become a new task in TickTick. Would that also be possible?

**Reply by NaturallyAsh** [reply]
If he were to implement that, I'd hope it'd be optional. Because I really like how its currently structured and would not want that one file per task setup in my vault. 

@thesamim  And I also prefer option 1.That syntax would work better with the Tasks Plugin. I actually don't think option 2 would be compatible with the Tasks Plugin, and It certainly be more work for you to code option 2 so that it could become compatible.

**Reply by thesamim** [reply]
> But there this a third option I would prefer even more: **A file per task**
> Meaning, each task in TickTick gets its own file in Obsidian. 

Interesting. What you would you want the file name to be? 

> Also, if I create a new file in Obsidian in a specified folder, it should become a new task in TickTick. Would that also be possible?

Tasks created in Obsidian, regardless of location, are created in TickTick if they have the #ticktick tag. If "Full Vault Sync" is turned on, all tasks created in Obsidian are created in Ticktick. So: What am I missing regarding your question?

**Reply by thesamim** [reply]
> @thesamim And I also prefer option 1.That syntax would work better with the Tasks Plugin. I actually don't think option 2 would be compatible with the Tasks Plugin, and It certainly be more work for you to code option 2 so that it could become compatible.

My testing so far: Both options are Tasks Plugin compatible. More testing to come, of course.

**Reply by Kageetai** [reply]
@NaturallyAsh of course, ideally optionally.

@thesamim I think the file name could just be the title of the task and the file content the description.
In regards to the file task creation, I meant that it would be great if the file itself constitutes the task, even without the `[ ]` if wanted. Therefore we could also later add checklist to the file, which would appear as checklists in the TickTick task as well.

Would that be feasible?

**Reply by thesamim** [reply]
@Kageetai :

>I think the file name could just be the title of the task and the file content the description. 

That's doable. There would have to be some rules around what the title can/cannot contain and the translation of characters that can't be in filenames....

>In regards to the file task creation, I meant that it would be great if the file itself constitutes the task, even without the `[ ]` if wanted. Therefore we could also later add checklist to the file, which would appear as checklists in the TickTick task as well.

That would require a completely different way of tracking tasks in the vault. I'm not saying it's not possible, I'm saying that it would be further down on the priority list than getting two way sync of notes. 

To clarify a bit:

1. In TickTick, a "task" can be a "Task" or a "Note". 
2. A "Note" task can only have markdown in it's content. 
3. A "Task"  task can have markdown or listitems in  it's content. 
4. Listitems are tracked by TickTick and can have reminders. 
5. Markdown content can also contain checklists, but they are not listitems. Meaning they can't have reminders.
6. As of now, and for the foreseeable future, TickTickSync does not keep track of listitem date elements anway, so the difference in terms of synchronization is not strictly relevant.
7. Sub-tasks are better for things that are part of a larger task and need to be date tracked.

Example of "Note" task:
![image](https://github.com/user-attachments/assets/0a9287e9-4f4f-4b39-aa67-979938e203e4)

Example of "Task" task, with list items:
![image](https://github.com/user-attachments/assets/6bd1b9bc-5f88-4f87-9b04-358eb9c69724)

Example of "Task" task, with content:
![image](https://github.com/user-attachments/assets/b0bfdae1-1514-44ad-856b-d956c7888fcb)

Hope this makes sense.

**Reply by Kageetai** [reply]
> That's doable. There would have to be some rules around what the title can/cannot contain and the translation of characters that can't be in filenames....

That makes sense of course. 👍 



> That would require a completely different way of tracking tasks in the vault. I'm not saying it's not possible, I'm saying that it would be further down on the priority list than getting two way sync of notes.

I understand and realise it's significantly more effort. It was just idea for the future, that would be cool to have. Thanks for considering it.

Just one idea maybe: If a file in Obsidian is related to a specific task in TickTick, just ignore any task tracking inside that file? And from TickTicks side, just parse the task description I to the file, ignoring any other list items, at least for the time being?!

**Comment by thesamim** [comment]
Howdy folks, 

**EDIT: Do NOT add comments to the issue on the Tasks plugin. They are already aware and have it on the priority list.**

Bottom line: Notes inline with Tasks is doable. The Tasks plugin affects reading view. Let me know if you have any other suggestions.

Detail:
Getting closer to implementing this feature. As it turns out, both options (indented lines, call out below tasks) are feasible and can be presented as a setting preference. (let me know which one you want the default to be.)

Bit of an issue the [Tasks](https://github.com/obsidian-tasks-group/obsidian-tasks) plugin does not support multi-line tasks (See: https://publish.obsidian.md/tasks/Getting+Started/Getting+Started#Multi-line+checklist+items), and this issue: https://github.com/obsidian-tasks-group/obsidian-tasks/issues/2061

From a practical stand point that means:

1. in live preview and source mode, all is well.
2. Tasks queries will only consider the first line of the task.
3. in reading mode, because of the DOM manipulation that the Tasks plugin does:
	3.1. indented notes disappear
	3.2. Callouts appear above the task

To see what I'm talking about, drop the following file into your vault. Go to reading mode. Disable the Tasks plugin, note appearance. Enable the Tasks plugin, note appearance.
[Tasks with Notes.md](https://github.com/user-attachments/files/18623084/Tasks.with.Notes.md)

NB: TickTickSync does not affect any of this. To prove the point disable TickTickSync....

**Comment by ms3056** [comment]
This is what I see in Live Preview without the Tasks plugin enabled - it looks like the only issue is the query, which is expected. Reading mode looks the same for me as well. 

![image](https://github.com/user-attachments/assets/6a7a6d9b-b31e-4409-a9aa-9a048d406d3c)

I am curious though - I was after the NOTES section of TickTick, not notes in tasks:

![image](https://github.com/user-attachments/assets/e249cd71-d66f-45e3-abba-0adf99bb6e05)

**Reply by thesamim** [reply]
"Notes" are no different than regular tasks. Meaning: They are currently being synched, but their content (the note you would write in the note section of the Note) is not. When I get done with the Notes work, they will be.
